packages/react-app/components/TokenContract.tsx

```diff
@@ -0,0 +1,489 @@
+import { useInput } from ".";
+import { useContractKit } from "@celo-tools/use-contractkit";
+import { useEffect, useState } from "react";
+import { useSnackbar } from "notistack";
+import { truncateAddress } from "../utils";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Divider from "@mui/material/Divider";
+import Grid from "@mui/material/Grid";
+import Typography from "@mui/material/Typography";
+import Link from "@mui/material/Link";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import IconButton from "@mui/material/IconButton";
+import SvgIcon from "@mui/material/SvgIcon";
+import { ethers } from "ethers";
+
+function HomeIcon(props) {
+  return (
+    <SvgIcon {...props}>
+      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
+    </SvgIcon>
+  );
+}
+
+const iconStyle = {
+  // backgroundColor: "#000000", // camel cased
+  // height: "100px",
+  // color: "yellow",
+  marginLeft: "0px",
+  marginRight: "0px",
+  marginTop: "10px",
+  marginBottom: "10px",
+};
+
+const buttonStyle = {
+  // backgroundColor: "#000000", // camel cased
+  // height: "100px",
+  // color: "yellow",
+  marginLeft: "0px",
+  marginRight: "5px",
+  marginTop: "20px",
+  marginBottom: "20px",
+};
+
+export function TokenContract({ contractData }) {
+  const { kit, address, network, performActions } = useContractKit();
+  const [tokenValue, setTokenValue] = useState();
+  const [tokenInput, setTokenInput] = useInput({ type: "text" });
+  const [contractLink, setContractLink] = useState("");
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+
+  //   const [{ data, error, loading }, getSigner] = useSigner();
+  const [tokenBalance, setTokenBalance] = useState("");
+  const [tokenName, setTokenName] = useState("");
+  const [tokenDecimals, setTokenDecimals] = useState("");
+  const [tokenInitialSupply, setTokenInitialSupply] = useState("");
+  const [tokenSymbol, setTokenSymbol] = useState("");
+  const [tokenTotal, setTokenTotal] = useState("");
+  const [tokenBalanceOf, setTokenBalanceOf] = useState("");
+  const [newAddress, setNewAddress] = useState("");
+  const [tokenTransferValue, setTokenTransferValue] = useState("");
+
+  const contract = contractData
+    ? new kit.web3.eth.Contract(contractData.abi, contractData.address)
+    : null;
+
+  useEffect(() => {
+    if (contractData) {
+      setContractLink(`${network.explorer}/address/${contractData.address}`);
+    }
+  }, [network, contractData]);
+
+  const setToken = async () => {
```

Comment:
my text editor is telling me that this function is never referenced. Should it be called somewhere?

Reply:
Looks like this is what I needed for `transferToken`. Fixed the syntax using this code so it works now.